src/Microsoft.OData.Core/ODataObjectModelExtensions.cs

```diff
@@ -34,6 +34,17 @@ public static void SetSerializationInfo(this ODataResourceBase resource, ODataRe
             resource.SerializationInfo = serializationInfo;
         }
 
+        /// <summary>
+        /// Provide additional serialization information to the <see cref="ODataWriter"/> for <paramref name="resource"/>.
+        /// </summary>
+        /// <param name="resource">The instance to set the serialization info.</param>
+        /// <param name="serializationInfo">The serialization info to set.</param>
+        public static void SetSerializationInfo(this ODataResource resource, ODataResourceSerializationInfo serializationInfo)

```

Comment:
>public static void SetSerializationInfo [](start = 8, length = 39)

The whole method body is to assign the value to its property. 
So, what's the advantages to add such extension method?

Can we just make the internal property 'SerializationInfo'  as public?

Reply:
I honestly don't know why this extension method was created in the first place, but it's shipped so we need to retain it. Can revisit in 8.x.

Reply:
I honestly don't know why this extension method was created in the first place, but it's shipped so we need to retain it. Can revisit in 8.x.

---
In reply to: [160273548](https://github.com/OData/odata.net/pull/1025#discussion_r160273548) [](ancestors = 160273548)